src/utils/git.ts

```diff
@@ -0,0 +1,71 @@
+import { exec } from '@actions/exec';
+import { logger } from './config';
+
+/* *
+ * GitHubのowner名やrepo名が正しくない場合はエラーにする
+  * @param value GitHubのowner名やrepo名
+  * @param options オプション
+  * @param options.name GitHubのowner名やrepo名の名前
+  * @param options.throwOnFail エラーが発生した場合に例外をスローするかどうか
+  * @returns 正しい場合はtrue、正しくない場合はfalse
+  */
+const validateGitHub = (value: string, options?: { name?: string, throwOnFail?: boolean }) => {
+  const pattern = /^[a-zA-Z0-9-]+$/;
+  const result = pattern.test(value);
+
+  if (!result) {
+    let msg = ''
+    if (options?.name) {
+      msg = `Invalid GitHub ${options.name} name: ${value}`;
+    } else {
+      msg = `Invalid GitHub name: ${value}`;
+    }
+
+    if (options?.throwOnFail) { throw new Error(msg); }
+    console.debug(msg);
+
+    throw new Error(msg);
+  }
+
+  return result;
+}
+
+export async function initializeGitConfig() {
+  await exec('git', ['config', '--global', 'user.name', 'github-actions']);
+  await exec('git', ['config', '--global', 'user.email', 'actions@github.com']);
+}
+
+export async function initializeUpstream(token: string, owner?: string, repo?: string) {
+  if (owner == null || repo == null) {
+    throw new Error("Upstream owner and repo must be specified or provided through a fork relationship.");
+  }
+
+  validateGitHub(owner, { name: 'owner', throwOnFail: true });
+  validateGitHub(repo, { name: 'repo', throwOnFail: true });
+
+  const upstreamUrl = `https://github.com/${owner}/${repo}.git`;
+
+  // 既にupstreamが存在する場合を考慮して、一度削除してから追加することで上書きする
+  await exec('git', ['remote', 'remove', 'upstream'], { ignoreReturnCode: true });
+  await exec('git', ['remote', 'add', 'upstream', upstreamUrl]);
```

Comment:
@coderabitai errorが起こっただけでは存在しないかどうかは断言できないから`"No existing upstream to remove."`とは断定できなくてFailed to remove branch upstream. This may be due to no existing upstream to remove.のように書くべきじゃない？

Reply:
@coderabbitai errorが起こっただけでは存在しないかどうかは断言できないから`"No existing upstream to remove."`とは断定できなくてFailed to remove branch upstream. This may be due to no existing upstream to remove.のように書くべきじゃない？

Reply:
code suggestionにして

Reply:
@coderabbitai code suggestionにして